src/SDK.ts

```diff
@@ -726,6 +731,110 @@ export default class SDK implements ISDK {
     });
   }
 
+  public async createConversation(requestId: string, sessionInitOptionalParams: ISessionInitOptionalParams = {}): Promise<FetchChatTokenResponse> {
+    const timer = Timer.TIMER();
+    this.logWithLogger(LogLevel.INFO, OCSDKTelemetryEvent.SESSIONINITSTARTED, "Session Init V2 Started", requestId);
+    const axiosInstance = axios.create();
+    const retryOn429 = true;
+    axiosRetryHandler(axiosInstance, {
+      retries: this.configuration.maxRequestRetriesOnFailure,
+      waitTimeInMsBetweenRetries: this.configuration.waitTimeBetweenRetriesConfig.sessionInit,
+      shouldRetry: (error) => sessionInitRetryHandler(error, retryOn429),
+      headerOverwrites: [OmnichannelHTTPHeaders.authCodeNonce]
+    });
+
+    const { reconnectId, authenticatedUserToken, initContext, getContext } = sessionInitOptionalParams;
+    const data: InitContext = initContext || {};
+    const requestHeaders: StringMap = { ...Constants.defaultHeaders };
+
+    const basePath = authenticatedUserToken
+      ? OmnichannelEndpoints.LiveChatConnecterAuthPath
+      : OmnichannelEndpoints.LiveChatConnecterPath;
+
+    const requestPath = `/${basePath}/${this.omnichannelConfiguration.orgId}/widgetApp/${this.omnichannelConfiguration.widgetId}/conversation`;
+
+    if (authenticatedUserToken) {     
+      Object.assign(requestHeaders, {
+        [OmnichannelHTTPHeaders.authenticatedUserToken]: authenticatedUserToken,
+        [OmnichannelHTTPHeaders.authCodeNonce]: this.configuration.authCodeNonce,
+      });
+    }
+
+    this.setSessionIdHeader(this.sessionId, requestHeaders);
+    addOcUserAgentHeader(this.ocUserAgent, requestHeaders);
+    this.seRequestIdHeader(requestId, requestHeaders);
+
+    if (reconnectId) {
+      data.reconnectId = reconnectId;
+    }
+
+    data.channelId = this.omnichannelConfiguration.channelId;
+
+    if (getContext && !window.document) {
+      return Promise.reject(new Error(`getContext is only supported on web browsers`));
+    }
+
+    if (getContext) {
+      data.browser = BrowserInfo.getBrowserName();
+      data.device = DeviceInfo.getDeviceType();
+      data.originurl = window.location.href;
+      data.os = OSInfo.getOsType();
+    }
+
+    // Set default locale if locale is empty
+    if (!data.locale) {
+      data.locale = Constants.defaultLocale;
+    }
+
+    // Validate locale
+    if (data.locale && !Locales.supportedLocales.includes(data.locale)) {
+      return Promise.reject(new Error(`Unsupported locale: '${data.locale}'`));
+    }
+
+    const url = `${this.omnichannelConfiguration.orgUrl}${requestPath}`;
+    const method = "POST";
+    const options: AxiosRequestConfig = {
+      data,
+      headers: requestHeaders,
+      method,
+      url,
+      timeout: this.configuration.defaultRequestTimeout ?? this.configuration.requestTimeoutConfig.sessionInit
+    };
+
+    return new Promise(async (resolve, reject) => {
```

Comment:
Can we remove `return new Promise()`? The method is an `async` function already which returns a promise by default. Now we're returning a promise inside a promise which makes it redundant